mmrazor/models/pruners/structure_pruning.py

```diff
@@ -98,9 +111,17 @@ def prepare_from_supernet(self, supernet):
 
         # record the visited module name during trace path
         visited = dict()
+        # shared modules such as shared detection head in RetinaNet
+        self.shared_module = []
```

Comment:
Add more detailed doc-strings to explain what is `shared_module` and why there need to record `shared_module`